### PR TITLE
[asc, sharecode]fix to intellij's requirements

### PR DESCRIPTION
**Pull request by wangmingliang-ms**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/IArtifact.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+package com.microsoft.azure.toolkit.lib.common.model;
+
+import lombok.Getter;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+
+public interface IArtifact {
+    String getId();
```
Comment (shenqianjin) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/IArtifact.java:15]: why not a super POJO ? I feel that "id" is only a property of a object.

Comment (wangmingliang-ms) [reply azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/IArtifact.java:15]: artifact is not a azure resource like `mysql` or `webapp`/`function`. 
and I dont see the necessity to abstract them at least for now and we may do this abstraction in the future when needed.

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/IArtifact.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Nonnull
+    static FileArtifact fromFile(@Nonnull final File file) {
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/IArtifact.java:18]: IMO, library just need to define the Artifact interface and tooling will provide the imlpements as well as the from method

Comment (Flanker32) [reply azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/IArtifact.java:18]: Besides, move this method to `FileArtifact`? I think we should not use the implementation of interface in interface itself.

File: azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/config/SpringCloudAppConfig.java [hunk 3]
```diff
@@ -6,22 +6,30 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
 import org.apache.commons.lang3.BooleanUtils;
 
-@Getter
 @Builder
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/config/SpringCloudAppConfig.java:20]: Why we need both `NoArgsConstructor` and `AllArgsConstructor` ?

**Pull request merged**